extensions/libarchive/MergeContent.h

```diff
@@ -259,6 +261,42 @@ class ZipMerge: public ArchiveMerge, public MergeBin {
   }
 };
 
+class AttributeMerger {
+public:
+  AttributeMerger(core::ProcessSession *session, std::deque<std::shared_ptr<org::apache::nifi::minifi::core::FlowFile>> &flows, std::shared_ptr<core::FlowFile> &mergeFlow)
+    : session_(session)
+    , flows_(flows)
+    , mergeFlow_(mergeFlow) {}
+  void mergeAttributes();
+  virtual ~AttributeMerger() = default;
+protected:
+  std::map<std::string, std::string> getCommonAttributes();
+  virtual void processFlowFile(const std::shared_ptr<core::FlowFile> &flow, std::map<std::string, std::string>& commonAttributes) = 0;
+
+  std::deque<std::shared_ptr<core::FlowFile>> &flows_;
```

Comment:
How about using a simple `core::FlowFile*` pointer here?

Reply:
Maybe some fancy `gsl_not_null`

Reply:
Why would be the benefit using a raw pointer over shared_ptr& here?

Reply:
I take it back, I see that this is actually a reference to the `BinFiles::queue_` that probably owns the flowfiles (it is irrelevant for this PR if it should be shared or not).

As you only take `const &`-s to the elements, you should probably make the member `const &` as well.